packages/components/tests/integration/components/hds/table/index-test.js

```diff
@@ -243,18 +260,80 @@ module('Integration | Component | hds/table/index', function (hooks) {
       .hasText('Sorted by artist descending');
   });
 
+  test('it renders a custom sortedMessageText if supplied', async function (assert) {
+    setSortableTableData(this);
+    this.set('sortedMessageText', 'Melanie will sort it');
+
+    await render(hbsSortableTable);
+    assert.dom('#data-test-table caption').hasText('Melanie will sort it');
+  });
+
+  test('it renders both a custom caption and a custom sortedMessageText if supplied', async function (assert) {
+    setSortableTableData(this);
+    this.set('caption', 'A custom caption.');
+    this.set('sortedMessageText', 'Melanie will sort it!');
+
+    await render(hbsSortableTable);
+    assert
+      .dom('#data-test-table caption')
+      .hasText('A custom caption. Melanie will sort it!');
+  });
+
+  test('it uses a custom sort function if one is supplied', async function (assert) {
```

Comment:
@hashicorp/hds-engineering this is the test I'm struggling with right now. 

1. It seems kind of useless to write a test that just copies what's in the showcase, because the showcase rendering indicates that the custom function did actually work. Trying to find the right approach for what should be tested.
2. Brian mentioned using Sinon, do we want to add another package? Will it add value?
3. Should we even be testing this?

Really stuck here so feedback is welcome.

Reply:
Some considerations:
1. the showcase (in terms of testing) is mainly for visual regression testing, not functional testing which is done via integration and acceptance tests
2. I don't know about sinon, so I can't help with it (my instinct tells me there's a risk to go down a rabbit hole with it)
3. yes, I think so because it's a complex functionality, and we want to make sure that is covered as much as possible by automatic tests, so that if in the future if we have to touch the code of the `Table` we have some safeguard in place

My suggestion is:
- the code I saw this morning, was a good direction apart from the two things I mentioned in my PR: the sorting function provided to the `@columns` should be a function, and it should be a "sorting" function that returns +1/-1/0 depending on the arguments (you can see a hint of it [in the `sort-by` source code](https://github.com/DockYard/ember-composable-helpers/blob/62fb18314e563d70c6588d733df4d26fd4b14aa1/addon/helpers/sort-by.js#L33)) and for this you can use as model the function we've already written for the custom sorting in the showcase: https://github.com/hashicorp/design-system/blob/melsumner/HDS-1414-table-custom-sorting/packages/components/tests/dummy/app/controllers/components/table.js#L28-L38

If you want, I can pair with you and see how to get this test working.

Reply:
Sure, let's look at it again tomorrow, I think I managed to figure it out, but see what you think.

Reply:
Works for me. I'll leave it to you to decide if you want to go with the duplication of code, or use the initial implementation that leveraged the `setSortableTableData` method.

Reply:
Fixed together in pairing session